compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/draganddrop/DragAndDropNode.kt

```diff
@@ -138,20 +179,67 @@ internal class DragAndDropNode(
 
     // end Node
 
+    /**
```

Comment:
The changes in `commonMain` are big. It will add more difficulties during merge.

I suggest to create a CL to AOSP and merge it first.

If it is needed for iOS implementation, we can create a branch.

If we want to include it in 1.7, there is an alternative - creating a copy of it, and change only small portion of `commonMain`.

Reply:
It's already there. And I wrote about it in description:

> Sync internals with https://r.android.com/3197432

Regarding

>  and change only small portion of commonMain.

It's already only ui/internal portion of the change

Reply:
> It's already there. And I wrote about it in description:

Better to merge it first. Only if we really need, we can merge.

Disadvantages of merging it now:
- the CL will most probably be changed because of the review
- the `androidx-main` code can be changed during AOSP new features/bugfixes

To mitigate this, we can do a 1-week monitoring of the changes, and apply them to `jb-main`. But if we have too many such changes, it is hard to keep track of them. Also we rely on authors of these changes. If they will be unavailable/busy with other things, the changes in common will decay.

> It's already only ui/internal portion of the change

I meant making it event more smaller, like changing 1-3 lines of common by changing a type, which we can copy.

Reply:
>  and apply them to jb-main

Also it will be hard to apply, because the CL is against `androidx-main`, but `jb-main` is against `1.7` branch which was created somewhere in May

Reply:
> Also it will be hard to apply, because the CL is against androidx-main, but jb-main is against 1.7 branch which was created somewhere in May

If we ho this path (updating our changes in common), we need to do them in `integration` first, and then cherry-pick to `jb-main`

Reply:
But merging the CL is much easier path, just longer

Reply:
Discussed online, decided to merge